### PR TITLE
More permissive parser for `env_keep` and friends.

### PR DESCRIPTION
**Pull request by squell**
Closing #384

### CODE REVIEW

File: src/sudoers/tokens.rs [hunk 1]
```diff
@@ -194,7 +194,12 @@ impl Token for EnvVar {
     }
 
     fn accept(c: char) -> bool {
-        c.is_ascii_alphanumeric() || "*_%".contains(c)
+        !c.is_control() && !c.is_whitespace() && !Self::escaped(c)
+    }
+
+    const ESCAPE: char = '\\';
+    fn escaped(c: char) -> bool {
+        "\\=#\"".contains(c)
```
Comment (japaric) [inline src/sudoers/tokens.rs:212]: nit: this is a compact way to write this but I found it a bit hard to parse out the four characters in the string because of the escaping. I think `matches!(c, '\\' | '=' | '#' | '"')` would be easier to parse, even if it's longer to type

**Review by japaric — APPROVED** [review]
left a suggestion but otherwise LGTM

**Comment by squell** [comment]
cc #514
